pkg/webhook/validation/uservalidation.go

```diff
@@ -38,6 +38,21 @@ func ValidateUserForFleetCR(ctx context.Context, client client.Client, whiteList
 	return slices.Contains(identities, userInfo.Username)
 }
 
-func isMasterGroupUserOrWhiteListedUser(whiteListedUsers []string, userInfo authenticationv1.UserInfo) bool {
+// ValidateUserForResource checks to see if user is allowed to modify argued fleet resource.
+func ValidateUserForResource(whiteListedUsers []string, userInfo authenticationv1.UserInfo, resKind, resName, resNamespace string) admission.Response {
+	if IsMasterGroupUserOrWhiteListedUser(whiteListedUsers, userInfo) || IsUserAuthenticatedServiceAccount(userInfo) {
+		klog.V(2).InfoS("user in groups is allowed to modify fleet resource", "user", userInfo.Username, "groups", userInfo.Groups, "kind", resKind, "name", resName, "namespace", resNamespace)
+		return admission.Allowed(fmt.Sprintf("user: %s in groups: %v is allowed to modify fleet resource %s: %s/%s", userInfo.Username, userInfo.Groups, resKind, resName, resNamespace))
+	}
+	return admission.Denied(fmt.Sprintf("user: %s in groups: %v is not allowed to modify fleet resource %s: %s/%s", userInfo.Username, userInfo.Groups, resKind, resName, resNamespace))
+}
+
+// IsMasterGroupUserOrWhiteListedUser returns true is user belongs to white listed users or user belongs to system:masters group.
+func IsMasterGroupUserOrWhiteListedUser(whiteListedUsers []string, userInfo authenticationv1.UserInfo) bool {
 	return slices.Contains(whiteListedUsers, userInfo.Username) || slices.Contains(userInfo.Groups, mastersGroup)
 }
+
+// IsUserAuthenticatedServiceAccount returns true if user is a valid/authenticated service account.
+func IsUserAuthenticatedServiceAccount(userInfo authenticationv1.UserInfo) bool {
+	return slices.Contains(userInfo.Groups, serviceAccountsKubeSystemGroup) && slices.Contains(userInfo.Groups, serviceAccountsGroup) && slices.Contains(userInfo.Groups, authenticatedGroup)
```

Comment:
Hi Arvind! This is an AND connected condition, but

any service account under the kube-system namespace will automatically be in the system:serviceaccounts group and system:authenticated group, so checking for system:serviceaccounts:kube-system alone should suffice

Reply:
And any service account will automatically be in the system:authenticated group, so checking for system:serviceaccounts group should suffice

Reply:
Is the purpose of this function to allow only service accounts in the kube-system namespace?